Source/WebKit/ModelProcess/cocoa/ModelProcessModelPlayerProxy.mm

```diff
@@ -90,10 +107,247 @@
 
 void ModelProcessModelPlayerProxy::loadModel(Ref<WebCore::Model>&& model, WebCore::LayoutSize layoutSize)
 {
-    RELEASE_LOG(ModelElement, "%p - ModelProcessModelPlayerProxy::loadModel size=%zu id=%" PRIu64, this, model->data()->size(), m_id.toUInt64());
+    // FIXME: Change the IPC message to land on load() directly
+    load(model, layoutSize);
+}
+
+static inline simd_float2 makeMeterSizeFromPointSize(CGSize pointSize, CGFloat pointsPerMeter)
+{
+    return simd_make_float2(pointSize.width / pointsPerMeter, pointSize.height / pointsPerMeter);
+}
+
+static simd_float3 computeExtents(simd_float2 boundsOfLayerInMeters, simd_float3 originalBoundingBoxExtents)
+{
+    if (simd_reduce_min(originalBoundingBoxExtents) - FLT_EPSILON > 0) {
+        auto boundsScaleRatios = simd_make_float2(
+            boundsOfLayerInMeters.x / originalBoundingBoxExtents.x,
+            boundsOfLayerInMeters.y / originalBoundingBoxExtents.y
+        );
+        return simd_reduce_min(boundsScaleRatios) * originalBoundingBoxExtents;
+    }
+
+    return originalBoundingBoxExtents;
+}
+
+static CGFloat degreesToRadians(CGFloat degrees)
+{
+    return degrees * 3.14159 / 180;
```

Comment:
M_PI?

Reply:
```
constexpr inline double deg2rad(double d)  { return d * radiansPerDegreeDouble; }
constexpr inline double rad2deg(double r)  { return r * degreesPerRadianDouble; }
```
in MathsExtras.h! Thanks for pointing out.